PopPUNK 2.6.0 - Improve `test` Stage

HarryHung opened the pull request:
As mentioned in #612 , this PR improve the `test` stage of Dockerfile of PopPUNK 2.6.0

In the `test` stage,  new files `test.sh` and `ftps.txt` are copied,
then `test.sh` is run to carry out the following actions:
1. Download 100 S. Pneumo assemblies from GPS Public Data on ENA
2. Build PopPUNK database from the assemblies
3. Assign clusters on the same assemblies using the built database
4. Compare the database clusters and assigned clusters of the assemblies

Note:

- I assume `test` stage is not used in the final published image, therefore no clean up is included in the `test` stage.

---

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

kapsakcj commented on poppunk/2.6.0/test.sh:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
```
What do you think about adding `set -euxo pipefail` somewhere near the top of this script? It makes the output more verbose, but it helps show exactly what is happening in the tests & makes them more robust if there were to be a failure

HarryHung replied:
Done! And good to know, as I am not familiar with automated testing.

kapsakcj commented:
One other request - could you please change line 13 to: `LABEL dockerfile.version="2"`

HarryHung commented:
Added `set -euxo pipefail`,
Changed the `LABEL dockerfile.version`
I could build the image on my machine with the updated files without any issue. 

Thanks for confirming on the docker build target for deployment. 

Building the test was a fun exercise to me, I hope it is useful. : )

kapsakcj commented:
Great! Super useful, thanks again for adding this.

You can view the automated GitHub Actions test here https://github.com/StaPH-B/docker-builds/actions/runs/4325784309/jobs/7552389191 and the "Build to test" section is where you can see the output of the `docker build` process

I will approve this PR and re-deploy the docker image, overwriting the same docker image tags since the `app` stage has not changed since yesterday & we are only adding a testing layer. Typically I'd prefer not to overwrite tags, but I think it's OK in this situation.

kapsakcj commented:
You can view the deployment workflow here, again it shouldn't take to long for the updated image to appear on dockerhub and quay: https://github.com/StaPH-B/docker-builds/actions/runs/4326024722

HarryHung commented:
Thanks for all the details. Highly appreciated.
